### PR TITLE
Derive FiniteGP from AbstractMvNormal

### PR DESCRIPTION
**Pull request by andreaskoher**
This PR fixes [issue #122](https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/issues/122). 

The only change is that FiniteGP is now derived from AbstractMvNormal instead of ContinuousMultivariateDistribution. 
Now, we can now directly sample from a latent finite GP in Turing as follows:

```julia
using AbstractGPs, KernelFunctions
using Turing

X = 1:100
f = GP( Matern32Kernel() )
u = rand( f(X) )
λ = exp.(u)
y = rand.(Poisson.(λ))

@model function latent_gp_model(X, y)
    f  = GP(Matern32Kernel())
    u ~ f(X)
    λ = exp.(u)
    y .~ Poisson.(λ)
end
c = sample( latent_gp_model(X, y), NUTS(), 10)
```

It's a somewhat trivial change, so I have not added any tests. Is there anything I missed?

### CODE REVIEW

**Comment by devmotion** [comment]
Can you maybe add the Turing example to the Turing tests?

**Comment by willtebbutt** [comment]
Do we want to consider this to be breaking? I guess stuff could in principle break as a consequence of this...

**Comment by devmotion** [comment]
Also related: https://github.com/JuliaGaussianProcesses/AbstractGPs.jl/issues/124

**Comment by andreaskoher** [comment]
> Can you maybe add the Turing example to the Turing tests?

ah right, will do.

File: test/turing.jl [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+			u ~ f(X)
+			λ = exp.(u)
+			y .~ Poisson.(λ)
+		end
+		m = latent_gp_regression(y, RowVecs(X))
+		@test length(sample(m, NUTS(), 5)) == 5
+		# Test for vectors of vector
+		m = latent_gp_regression(y, x)
+		@test length(sample(m, NUTS(), 5)) == 5
+	end
```
Comment (devmotion) [inline test/turing.jl:43]: I think we should avoid tabs:
```suggestion
        X = randn(3, 1)
        x = [rand(1) for _ in 1:3]
        y = rand.(Poisson.(exp.(randn(3))))

        @model function latent_gp_regression(y, X)
            f  = GP(Matern32Kernel())
            u ~ f(X)
            λ = exp.(u)
            y .~ Poisson.(λ)
        end
        m = latent_gp_regression(y, RowVecs(X))
        @test length(sample(m, NUTS(), 5)) == 5
        # Test for vectors of vector
        m = latent_gp_regression(y, x)
        @test length(sample(m, NUTS(), 5)) == 5
    end
```

**Comment by devmotion** [comment]
It seems the only thing left to do is to update the version number?

**Comment by willtebbutt** [comment]
Do we want to clean out our deprecations before releasing 0.3, or leave them in until 0.4? They've not been around for a particularly large amount of time, so possibly we want to keep them around for a bit?

**Comment by devmotion** [comment]
Hmm I don't know, I guess it's your call: according to JuliaHub only your packages depend on AbstractGPs (https://juliahub.com/ui/Packages/AbstractGPs/NOFPh/0.2.25?t=2) and I assume that not (too) many users work with it directly yet.

**Comment by willtebbutt** [comment]
Great, lets get rid of the deprecations then -- I've already upgraded Stheno anyway 🤷

**Review by willtebbutt — APPROVED** [review]
LGTM.

**Comment by devmotion** [comment]
Thank you for your contribution, @andreaskoher!

**Pull request merged**